dist/READMEcontainer.md

```diff
@@ -0,0 +1,70 @@
+ovn container construction and operation
+
+NOTES/LIMITS:
+- This only works on a cluster with a single master. Openshift
+  supports multiple masters.
+- Single network cidr only, Openshift supports a list of cidrs.
+- ovs daemon set is expected to be running. Its not part of ovn.
+- Uses node-role.kubernetes.io/master and compute labels. 
+  Eventually there will be ovn specific labels.
+- currently building in ../go_controller and copying files to
+  ../images and put in an image. This will likely change.
+NOTES/LIMITS:
+
+There are two daemonsets that support ovn. ovnkube-master runs
+on the cluster master node, ovnkube runs on the remaining nodes.
+The daemonsets run with hostNetwork: true.
+
+The both daemonsets run the node daemons, ovn-controller and ovn-node.
+In addition the daemonset runs ovn-northd and ovn-master.
+
+The startup sequence requires this startup order:
+- ovs (another daemonset starts this)
+- ovnkube-master on the master node
+- ovnkube on the rest of the nodes.
+
+===============================
+
+There is one docker image that is used by both the ovnkube-master
+and ovnkube daemonsets. The master sets environmet variable
+OVN_MASTER="true".
+
+When this is present the image runs ovn-northd and ovn-master
+before running ovn-controller and ovn-node
+ 
+The image entrypoint is /root/ovnkube.sh
+This script sequences through the operations to bring up networking.
+Configuration is passed in via environment variables.
+
+===============================
+
+- ovnkube-master.yaml
+
+    metadata:
+      labels:
+        app: ovnkube-master
+        node-role.kubernetes.io/master: "true"
+
+- ovnkube.yaml
+
+    metadata:
+      labels:
+        app: ovnkube
+        node-role.kubernetes.io/compute: "true"
+
+===========================
+How networking is set up.
+
+The ovn cni plugin, ovn-k8s-cni-overlay, must be visible to docker so that 
```

Comment:
Shouldn't "docker" here instead be "kubelet"?  We just need to be sure that kubelet can run the CNI plugin directly, it doesn't matter where that plugin is whether it's on the host filesystem or in a container that's mapped to the host filesystem.

Reply:
OK